zookeeper/src/main/java/com/linecorp/armeria/client/zookeeper/DiscoverySpec.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.zookeeper;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.server.zookeeper.InstanceSpec;
+
+/**
+ * A discovery specification for {@link ZooKeeperEndpointGroup}. The specification is used for finding
+ * and decoding the registered instances into {@link Endpoint}s.
+ *
+ * @see InstanceSpec
+ */
+public interface DiscoverySpec {
+
+    /**
+     * Returns the default {@link DiscoverySpec} implementation which assumes a zNode value is
+     * a comma-separated string. Each element of the zNode value represents an {@link Endpoint} whose format is
+     * {@code <host>[:<port_number>[:weight]]}, such as:
+     * <ul>
+     *   <li>{@code "foo.com"} - default port number, default weight (1000)</li>
+     *   <li>{@code "bar.com:8080} - port number 8080, default weight (1000)</li>
+     *   <li>{@code "10.0.2.15:0:500} - default port number, weight 500</li>
+     *   <li>{@code "192.168.1.2:8443:700} - port number 8443, weight 700</li>
+     * </ul>
+     * Note that the port number must be specified when you want to specify the weight.
+     */
+    static DiscoverySpec ofDefault() {
```

Comment:
default or legacy?

Reply:
Let's use legacy because I think we should guide people to use curator x which is a de-facto standard. 😉